### DISCUSSION TITLE
Redirect page after amount of time

### DISCUSSION THREAD

**Discussion by Lord-Y** [bigskysoftware/htmx#2135] [Ideas]
Hello guys,

After receiving an HTTP request 200 from the backend, we cannot redirect the actual page to another after some delay.
We need to use `HX-Trigger-After-Swap` and write an event with it which is kind of too much just for a redirect.

On the documentation [here](https://htmx.org/reference/#response_headers) we have `HX-Redirect` but we should also have `HX-Redirect-Timeout` header to make the job done in a more cleaner way.

Let me know what you think.

**Comment by svenberkvens** [comment]
I don't understand the problem. What makes it such that you can only redirect after a specific amount of time has passed?

**Reply by Lord-Y** [reply]
@svenberkvens To be able to print a message first on user side and then redirect when the delay is reached.

**Reply by efosao** [reply]
You have many options that will get you the same results.
Here are just a few off the top of my head...

- You can return a script tag as part of your HTML response
  - ```js<script>{`setTimeout(() => { window.location.href = "..."; }, 5000);`}</script>```
- You can you use `htmx:afterSettle` or `htmx:afterSwap` to call a function that triggers the redirect after the delay.

**Reply by Lord-Y** [reply]
@efosao That what I did. My point was to add `HX-Redirect-Timeout` header without having to write any `js` function.

**Reply by efosao** [reply]
> @efosao That what I did. My point was to add `HX-Redirect-Timeout` header without having to write any `js` function.

I like that is not a bloated framework. The smaller the better. A better name for what you're describing would be `HX-Redirect-Delay`. 

I definitely don't think it's necessary since you can still get the same functionality with a one line script tag, but that's just my personal opinion.

**Reply by svenberkvens** [reply]
How about this: when the action is trigged, you return a user message like this in a first fragment:

```
<div hx-get="/the/actual/redirect" hx-trigger="load delay:5s">
Some message for the user
</div>
```

And the `/the/actual/redirect` back-end call emits  the `HX-Redirect` header.

**Reply by efosao** [reply]
Even better. 👍

**Reply by Lord-Y** [reply]
Thx for the advices guys. I close it.

**Comment by gerdint** [comment]
Note that in HTMX 2.0.4 you need to add the `hx-target` attribute to the hx-get div for this to work (or you'll get an `htmx:targetError`).

**Comment by gerdint** [comment]
I didn't check, but I guess another way of solving this would be to add the [head support](https://htmx.org/extensions/head-support/) extension and add a meta refresh header with a content value of "5;url=/url/to/redirect/to". Possibly more onerous but should eliminate the need for the extra /the/actual/redirect endpoint
